### ISSUE TITLE
Asgard's Wrath is unflappably blurry 

### ISSUE THREAD

**Issue by cubiclestorm** [LibreVR/Revive#1353]
Valve Index
i7 8700k
Nvidia 2080ti

on all settings the game is extremely blurry. This is something to do with TAA as disabling it with an ini tweak fixes the issue. However turning off TAA gives you a black screen when setting graphics to High or Epic

**Comment by LuckMic** [comment]
I also have a Valve Index and I'm using SteamVR with SS 150%. Asgard's Wrath is not blurry on my PC with setting to High and Epic.

**Comment by mmorselli** [comment]
have you tried with this settings? 

`%appdata%\..\Local\WrathGame\Saved\Config\WindowsNoEditor\`

in engine.ini

```
[SystemSettings]

r.DefaultFeature.AntiAliasing=2
r.TemporalAACurrentFrameWeight=0.2
r.TemporalAASamples=16
r.TemporalAASharpness=1.0
r.Tonemapper.Sharpen=1.5
```

in scalibility.ini

```
[PostProcessQuality@3]
r.DefaultFeature.AntiAliasing=2
```

in gameusersettings.ini

`sg.PostProcessQuality=3`

**Comment by cubiclestorm** [comment]
Yes I have it's not good enough. Nowhere as sharp as when on Rift-S

**Comment by zxsdklfhskyrvk** [comment]
@LuckMic Care to share your steamvr config and your game config files?  As well as a hardware summary?  Do you have re-projection or stuttering?  Are you using fpsVR to determine performance or just eyeballing it?

**Comment by hookmanuk** [comment]
+1

On Index TAA is borderline unplayable it's so blurry. Rift S must be sharper than this as I can't believe it looks like this natively.

If so, then hopefully there's an issue with revive and TAA that can be improved?

**Comment by mmorselli** [comment]
I'm playing on Pimax 5K+ using its native Oculus support, because with ReVive I have too much stuttering, and with the above settings is so sharp that I don't think anyone could want more. So +1 to the theory that we need a ReVive specific support

**Comment by cubiclestorm** [comment]
> @LuckMic Care to share your steamvr config and your game config files? As well as a hardware summary? Do you have re-projection or stuttering? Are you using fpsVR to determine performance or just eyeballing it?

no vrfps. I'm getting good performance. no reprojection.

**Comment by cubiclestorm** [comment]
> +1
> 
> On Index TAA is borderline unplayable it's so blurry. Rift S must be sharper than this as I can't believe it looks like this natively.
> 
> If so, then hopefully there's an issue with revive and TAA that can be improved?

I have a Rift S and it works fine no blur

**Comment by LuckMic** [comment]
Valve Index
i7-6700
Nvidia 2080ti
SteamVR 90Hz, 150% (2488x2740), active motion-smoothing
In-game All default only changed graphics to high. No changes in the engine.ini or anything else.
Result: No stuttering and sharp. On epic I've sometimes stuttering but depends on the location. Reprojection is active running on around 10 ms (90 Hz).
BTW: I copied the game from my drive to a m.2 ssd. Loadings around 10 up to 20 seconds only!

**Comment by hookmanuk** [comment]
I've been playing mainly at 90, which presumably works fine on CV1.

I suspect people that think it is fine are just not realising how much clearer it can be without TAA. Fallout 4 & No Mans Sky both had the same issue I remember by default, TAA just kills all the pixel level details in VR.

I don't understand why the Rift S is so much sharper though (as you say it is), perhaps TAA is implemented better in the Rift SDK than the Steam VR SDK?

**Comment by cubiclestorm** [comment]
> I've been playing mainly at 90, which presumably works fine on CV1.
> 
> I suspect people that think it is fine are just not realising how much clearer it can be without TAA. Fallout 4 & No Mans Sky both had the same issue I remember by default, TAA just kills all the pixel level details in VR.
> 
> I don't understand why the Rift S is so much sharper though (as you say it is), perhaps TAA is implemented better in the Rift SDK than the Steam VR SDK?

TAA may be fixed to native resolution. I'm not surprised people think it's just fine. When I pointed out the canted display distortion when Index was new no one else saw it but it but me and the devs. People got some low standards or really bad eyes
